File: ProjetFinal/src/main/java/projetFinal/entities/Reservation.java
package projetFinal.entities;

import java.time.LocalDate;
import java.util.Objects;

public abstract class Reservation {
	private Long id_reservation;
	private Client client;
	private Restaurant restaurant;
	private LocalDate date = LocalDate.now();
	private String specification;
	
	
	public Reservation(Client client, Restaurant restaurant, LocalDate date, String specification) {
		super();
		this.client = client;
		this.restaurant = restaurant;
		this.date = date;
		this.specification = specification;
	}


	public Reservation() {
		super();
	}


	public Reservation(Long id_reservation, Client client, Restaurant restaurant, LocalDate date,
			String specification) {
		super();
		this.id_reservation = id_reservation;
		this.client = client;
		this.restaurant = restaurant;
		this.date = date;
		this.specification = specification;
	}


	public Long getId_reservation() {
		return id_reservation;
	}


	public void setId_reservation(Long id_reservation) {
		this.id_reservation = id_reservation;
	}


	public Client getClient() {
		return client;
	}


	public void setClient(Client client) {
		this.client = client;
	}


	public Restaurant getRestaurant() {
		return restaurant;
	}


	public void setRestaurant(Restaurant restaurant) {
		this.restaurant = restaurant;
	}


	public LocalDate getDate() {
		return date;
	}


	public void setDate(LocalDate date) {
		this.date = date;
	}


	public String getSpecification() {
		return specification;
	}


	public void setSpecification(String specification) {
		this.specification = specification;
	}


	@Override
	public int hashCode() {
		return Objects.hash(id_reservation);
	}


	@Override
	public boolean equals(Object obj) {
		if (this == obj)
			return true;
		if (obj == null)
			return false;
		if (getClass() != obj.getClass())
			return false;
		Reservation other = (Reservation) obj;
		return Objects.equals(id_reservation, other.id_reservation);
	}
	
}

File: projetFinalAngular/src/app/model/reservation.ts
import { Client } from "./client";
import { Restaurant } from "./restaurant";
import { Utilisateur } from "./utilisateur";

export abstract class Reservation {
  public get id(): number | undefined {
    return this._id;
  }
  public set id(value: number | undefined) {
    this._id = value;
  }

  public get client(): Utilisateur  {
    return this._client;
  }
  public set client(value: Utilisateur ) {
    this._client = value;
  }

  public get restaurant(): Restaurant | undefined {
    return this._restaurant;
  }
  public set restaurant(value: Restaurant | undefined) {
    this._restaurant = value;
  }

  public get date(): Date | undefined {
    return this._date;
  }
  public set date(value: Date | undefined) {
    this._date = value;
  }

  public get specification(): string | undefined {
    return this._specification;
  }
  public set specification(value: string | undefined) {
    this._specification = value;
  }

  public constructor(
    private _client: Utilisateur,
    private _id?: number,
    private _restaurant?: Restaurant,
    private _date?: Date,
    private _specification?: string
  ) {}

}
